File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int sum = 0;

        if (s.charAt(s.length() - 1) == '(') return false;
        
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            
            sum = c == '(' ? sum + 1 : sum - 1;
            if (sum == -1) return false;
        }
        
        answer = sum == 0 ? true : false;

        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let sum = 0;

    if (s[0] === ")" || s[s.length-1] === "(") return false;
    
    for (const str of s) {
        sum = str === "(" ? sum + 1 : sum - 1;
        
        if (sum < 0) return false;
    }

    return sum == 0 ? true : false;
}